snorkel/annotations.py

```diff
@@ -66,27 +66,30 @@ def _get_sliced_indexes(self, s, axis, index, inv_index):
                 return index, inv_index
             else:
                 idxs = np.arange(self.shape[axis])[s]
-        # csr_matrix._get_submatrix only handles slice or int, so this is int
-        else:
+        elif isinstance(s, int):
             idxs = np.array([s])
+        else:
+            # s is an array of ints
+            idxs = s
         index_new, inv_index_new = {}, {}
         for i_new, i in enumerate(idxs):
             k = index[i]
             index_new[i_new] = k
             inv_index_new[k] = i_new
         return index_new, inv_index_new
 
-    def _get_submatrix(self, row_slice, col_slice):
-        # Get the slice of the matrix
-        X = super(csr_AnnotationMatrix, self)._get_submatrix(row_slice,
-            col_slice)
-        X.annotation_key_cls = self.annotation_key_cls
-
-        # # Remap the row and column indexes
-        X.row_index, X.candidate_index = self._get_sliced_indexes(
-            row_slice, 0, self.row_index, self.candidate_index)
-        X.col_index, X.key_index = self._get_sliced_indexes(
-            col_slice, 1, self.col_index, self.key_index)
+    def __getitem__(self, key):
+        X = super(csr_AnnotationMatrix, self).__getitem__(key)
+
+        # Remap the row and column indexes if applicable.
+        if hasattr(X, 'row_index') and hasattr(X, 'col_index'):
```

Comment:
If you rely on the standard `__getitem__` method, I believe it will in some cases just return a generic `csr_matrix`, which will not have these attributes, e.g. see https://github.com/scipy/scipy/blob/v0.19.1/scipy/sparse/csr.py#L406.  Then the matrix returned won't have the row/col indexes...